File: src/handlers/cors.js
import {
    IncomingMessage,
    ServerResponse
} from 'node:http'

/**
 * @param {Array<string>} methods
 * @param {function(IncomingMessage, ServerResponse): void} posteriorHandler
 * @param {URL} url
 * @param {IncomingMessage} req
 * @param {ServerResponse} res
 */
const corsMiddleware = (methods, posteriorHandler, url, req, res) => {
    res.setHeader("Content-Type", "application/json; charset=utf-8")
    res.setHeader("Access-Control-Allow-Methods", methods.toString())
    res.setHeader("Access-Control-Allow-Origin", "*")
    res.setHeader("Access-Control-Allow-Headers", "*")

    posteriorHandler(req, res, url)
}

export {
    corsMiddleware
}

File: legacy_go/handlers/cors.go
package handlers

import "net/http"

func CorsMiddleware(handle http.Handler, methods string) http.HandlerFunc {
    return func(w http.ResponseWriter, r *http.Request){
        w.Header().Set("Content-Type", "application/json; charset=utf-8")
        w.Header().Set("Access-Control-Allow-Methods", methods)
        w.Header().Set("Access-Control-Allow-Origin", r.RemoteAddr)
        w.Header().Set("Access-Control-Allow-Headers", "*")

        if r.Method == http.MethodOptions {
            w.WriteHeader(http.StatusOK)
            return
        }

        handle.ServeHTTP(w, r)
    }
}

